File: agent-app-ktor/src/test/kotlin/com/riwonace/agentktor/answer/StructuredEvidenceFormatterTest.kt
package com.riwonace.agentktor.answer

import kotlin.test.Test
import kotlin.test.assertEquals

class StructuredEvidenceFormatterTest {
    private val formatter = StructuredEvidenceFormatter()

    @Test
    fun `명시된 count 단위를 보존한다`() {
        assertEquals("총 8개 입니다.", formatter.formatSql("고객사는 몇 개야?", "count=8"))
        assertEquals("총 4명 입니다.", formatter.formatSql("직원은 몇 명이야?", "count=4"))
    }

    @Test
    fun `명시 단위가 없으면 질문의 도메인 명사로 보완한다`() {
        assertEquals("총 5건 입니다.", formatter.formatSql("해결되지 않은 건은?", "count=5"))
        assertEquals("총 3곳 입니다.", formatter.formatSql("등록 고객사 수는?", "count=3"))
    }

    @Test
    fun `단일 count가 아닌 구조화 결과는 변경하지 않는다`() {
        val rows = "name=Client-A, count=3\nname=Client-B, count=2"
        assertEquals(rows, formatter.formatSql("고객사별 계약 수는?", rows))
    }
}

File: agent-app-go/internal/answer/structured_evidence_formatter_test.go
package answer

import "testing"

func TestStructuredEvidenceFormatter_PreservesExplicitCountUnit(t *testing.T) {
	f := &StructuredEvidenceFormatter{}
	if got := f.FormatSQL("고객사는 몇 개야?", "count=8"); got != "총 8개 입니다." {
		t.Fatalf("got %q", got)
	}
	if got := f.FormatSQL("직원은 몇 명이야?", "count=4"); got != "총 4명 입니다." {
		t.Fatalf("got %q", got)
	}
}

func TestStructuredEvidenceFormatter_NoExplicitUnitInfersFromDomainNoun(t *testing.T) {
	f := &StructuredEvidenceFormatter{}
	if got := f.FormatSQL("해결되지 않은 건은?", "count=5"); got != "총 5건 입니다." {
		t.Fatalf("got %q", got)
	}
	if got := f.FormatSQL("등록 고객사 수는?", "count=3"); got != "총 3곳 입니다." {
		t.Fatalf("got %q", got)
	}
}

func TestStructuredEvidenceFormatter_NonSingleCountResultUnchanged(t *testing.T) {
	f := &StructuredEvidenceFormatter{}
	rows := "name=Client-A, count=3\nname=Client-B, count=2"
	if got := f.FormatSQL("고객사별 계약 수는?", rows); got != rows {
		t.Fatalf("got %q", got)
	}
}
